lwcomponent/types.go

```diff
@@ -0,0 +1,50 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package lwcomponent
+
+type Type string
+
+const (
+	EmptyType Type = ""
+
+	// the component is a binary
+	BinaryType = "BINARY"
+
+	// will this component be accessible via the CLI
```

Comment:
I don't fully grok this component type.  It seems like other types (i.e. BINARY, STANDALONE, LIBRARY) are actual component types.  `CLI_COMMAND` seems like component behavior which isn't dictated by the component.

Reply:
I do agree with your comment, this component type is unique since it will be shown in the help message of the CLI and users can execute it via `lacework <cli_component>`. Maybe we should talk about other ways to indicate whether a component will be shown or not, I am open to suggestions.